src/conn.c

```diff
@@ -977,7 +959,8 @@ void conn_disconnect(xmpp_conn_t *conn)
         tls_free(conn->tls);
         conn->tls = NULL;
     }
-    sock_close(conn->sock);
+    if (conn->sock >= 0)
```

Comment:
On Windows sock_t is unsigned:
```
../src/conn.c: In function 'conn_disconnect':
../src/conn.c:962:20: warning: comparison of unsigned expression in '>= 0' is always true [-Wtype-limits]
  962 |     if (conn->sock >= 0)
      |                    ^~
```

suggest instead `if (conn->sock != INVALID_SOCKET)` and for non-Windows add `#define INVALID_SOCKET -1` inside `#ifndef _WIN32` in `sock.h`. (`INVALID_SOCKET` is already defined by Winsock)

Reply:
Will fix it,

Reply:
done